joins.qmd

```diff
@@ -484,7 +485,7 @@ The number of dots equals the number of matches, which in turn equals the number
 knitr::include_graphics("diagrams/join/inner.png", dpi = 270)
 ```
 
-An **outer join** keeps observations that appear in at least one of the data frames.
+We can apply the same principles to explain the **outer joins**, which keep observations that appear in at least one of the data frames.
```

Comment:
I don't know if this is valuable, but I've enjoyed thinking of outer joins as performing _exactly_ the same matching procedure as an inner join - the only difference is in how they handle _unmatched_ rows on either side of the join.

i.e. the base of an outer join is an inner join, patched up with some of the unmatched rows

Reply:
I guess this is different from how you talk about it though, because you say that there is a catch-all virtual row that matches against "anything" that isn't already handled. So that takes the place of talking about handling _unmatched_ rows, because in your explanation there aren't any anymore (because the catch all handled them already)

Reply:
I would keep the description as is but remind by appending ", unlike an inner join that keeps observations that appear in both tables."